### PR TITLE
Remove duplicate features

### PR DESCRIPTION
**Pull request by 545305338**
When looking for exact features, sub-pixel interpolation may occur with repetitive features

### CODE REVIEW

File: src/sift.c [hunk 1]
```diff
@@ -380,14 +384,29 @@ static CvSeq* scale_space_extrema( IplImage*** dog_pyr, int octvs, int intvls,
...
+                        {
+                          cvSeqPush( features, feat );
+                          feature_mat[dog_pyr[o][0]->width * ddata->r + ddata->c] += 1;
+                        }
+                        else if( ddata->intvl == 2 && (feature_mat[dog_pyr[o][0]->width * ddata->r + ddata->c] & 0x2) == 0 )
+                        {
+                          cvSeqPush( features, feat );
+                          feature_mat[dog_pyr[o][0]->width * ddata->r + ddata->c] += 2;
+                        }
+                        else if( ddata->intvl == 3 && (feature_mat[dog_pyr[o][0]->width * ddata->r + ddata->c] & 0x4) == 0 )
```
Comment (robwhess) [inline src/sift.c]: What if `ddata->intvl` is greater than 3?

**Comment by 545305338** [comment]
Thanks robwhess tell me the fragile code that I modified it to avoid duplicate features within 32 or 64 layers.
The attached image is extracted with repetitive features：79.075527 121.917857 1.549677 0.278847
crayfish

File: src/sift.c [hunk 2]
```diff
@@ -380,14 +383,21 @@ static CvSeq* scale_space_extrema( IplImage*** dog_pyr, int octvs, int intvls,
 		    if( ! is_too_edge_like( dog_pyr[ddata->octv][ddata->intvl],
 					    ddata->r, ddata->c, curv_thr ) )
 		      {
-			cvSeqPush( features, feat );
+                        if( ddata->intvl > sizeof(unsigned long) )
```
Comment (robwhess) [inline src/sift.c]: I think this needs to be `>=`.

Comment (545305338) [inline src/sift.c]: Unnecessary because the shift is ddata->intvl-1
right？

Comment (robwhess) [inline src/sift.c]: Oh, right.  Sorry, I missed that.

**Pull request merged**